packages/context/src/context.ts

```diff
@@ -135,14 +140,15 @@ export class Context extends EventEmitter {
    * as the prefix
    * @param args - Arguments for the debug
```

Comment:
What happens when the child class calls this method but does not provide a custom `_debug` logger? I believe your implementation will use the `debug` function initialized with the scope `loopback:context` (see line 43 above). As a result, debug logs printed by a custom context subclass may use `loopback:context` scope, which may be surprising to users.

At minimum, I'd like this aspect to be clearly described in API docs.

I think a better solution is to find a design that will make it impossible to call debug with the default scope. For example, the base `Context` class can provide a helper function accepting a context instance and a debug logger and returning a debug function that includes the context name in all messages. Then each Context subclass can use that helper to create its own private debug logger.

```ts
export function createDebugWithContextDetails(ctx: Context, debug: Debugger) {
  return function debugWithName(...args: unknown[]) {
    // put your code here, replace `this.name` with `ctx.name`
    // in the future, we can add more details in addition to the name
  }
 }

// inside Context
class Context {
  private _debug = createDebugWithContextDetails(this, debug);
  // ...
}

// in the core
import {createDebugWithContextDetails} from '@loopback/context';

class Application {
  private _debug = createDebugWithContextDetails(this, debug);
  // ...
}
```

If you prefer, the helper can be a static method on the Context class. I am sure we can find a better name too.

Reply:
We now use the subclass name to build the debug namespace by default. It can be also overridden with `_debug` or `getDebugNamespace`.